### ISSUE TITLE
[Bug]: sp-color-slider and sp-color-wheel fail to render focused style when first receiving focus

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3278]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

ColorSlider, ColorWheel

### Expected behavior

`<sp-color-slider>` and `<sp-color-wheel>` should render the `focused` style when they first receive focus, tabbing into the document of iframe.

### Actual behavior

For ColorSlider:

1. Open https://opensource.adobe.com/spectrum-web-components/components/color-slider/#default 
2. Hover over and mouse click to set focus on the "#" sign next to the heading reading "Default."
3. Press Tab key once to set focus to the "#" link with href="#default".
4. Then press Tab a second time to move focus to the ColorSlider example rendered below.
5. Notice that the ColorSlider handle does not change, by growing in size, to display focus.
6. If you press Esc or an Arrow key to change the value, the focused style will render.

Repeat for ColorWheel:
1. Open https://opensource.adobe.com/spectrum-web-components/components/color-wheel/#example 
2. Hover over and mouse click to set focus on the "#" sign next to the heading reading "Example."
3. Press Tab key once to set focus to the "#" link with href="#default".
4. Then press Tab a second time to move focus to the ColorWheel example rendered below.
5. Notice that the ColorWheel handle does not change, by growing in size, to display focus.
6. If you press Esc or an Arrow key to change the value, the focused style will render.

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error

### Sample code that illustrates the problem

In order to account for the focused element in the shadowRoot matching  `:focus-visible` or `.focus-visible`, change  this:
https://github.com/adobe/spectrum-web-components/blob/2682d937d392f9858ea757e42acfb8b4ca738753/tools/base/src/Base.ts#L81-L100

To this:
```ts
        public hasVisibleFocusInTree(): boolean {
            const activeElement = (this.getRootNode() as Document)
                .activeElement as HTMLElement;
            if (!activeElement) {
                return false;
            }

            const shadowRootActiveElement = activeElement.shadowRoot
                ?.activeElement as HTMLElement;

            // Browsers without support for the `:focus-visible`
            // selector will throw on the following test (Safari, older things).
            // Some won't throw, but will be focusing item rather than the menu and
            // will rely on the polyfill to know whether focus is "visible" or not.
            try {
                return (
                    activeElement.matches(':focus-visible') ||
                    activeElement.matches('.focus-visible') ||
                    shadowRootActiveElement?.matches(':focus-visible') ||
                    shadowRootActiveElement?.matches('.focus-visible')
                );
                /* c8 ignore next 3 */
            } catch (error) {
                return (
                    activeElement.matches('.focus-visible') ||
                    shadowRootActiveElement?.matches('.focus-visible')
                );
            }
        }
```
